### PR TITLE
Users/xinlian/included path and exclude path update

### PR DESCRIPTION
**Pull request by xinlian12**
This PR solve https://github.com/Azure/azure-sdk-for-java/issues/11192

IncludedPath ->

Remove constructor which takes jsonString
Add constructor which takes String path - and call logic of setPath(path) in the constructor
Change public Collection<Index> getIndexes() { -> public List<Index> getIndexes() 

ExcludedPath ->
Add constructor which takes String path - and call logic of setPath(path) in the constructor

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-benchmark/src/test/java/com/azure/cosmos/benchmark/ReadMyWritesConsistencyTest.java [hunk 1]
```diff
@@ -168,7 +168,7 @@ DocumentCollection getCollectionDefinitionWithRangeRangeIndex() {
         List<IncludedPath> includedPaths = new ArrayList<>();
         IncludedPath includedPath = new IncludedPath();
         includedPath.setPath("/*");
-        Collection<Index> indexes = new ArrayList<>();
+        List<Index> indexes = new ArrayList<>();
         Index stringIndex = Index.range(DataType.STRING);
         BridgeInternal.setProperty(ModelBridgeInternal.getJsonSerializable(stringIndex), "precision", -1);
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/src/test/java/com/azure/cosmos/benchmark/ReadMyWritesConsistencyTest.java:173]: can this be updated to use the public apis for setting precision.

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/models/IncludedPathTest.java [hunk 2]
```diff
@@ -1,40 +0,0 @@
...
-
-import java.util.ArrayList;
-import java.util.List;
-
-import static org.assertj.core.api.Assertions.assertThat;
-
-public class IncludedPathTest {
-
-    @Test(groups = {"unit"})
-    public void deserialize() {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/models/IncludedPathTest.java:16]: now that custructor for parsing json is removed, could you please add a test which sets different fields using direct api invocation?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ExcludedPath.java [hunk 3]
```diff
@@ -23,10 +23,11 @@ public ExcludedPath() {
     /**
      * Constructor.
      *
-     * @param jsonString the json string that represents the excluded path.
+     * @param path the excluded path.
      */
-    ExcludedPath(String jsonString) {
-        this.jsonSerializable = new JsonSerializable(jsonString);
+    public ExcludedPath(String path) {
```
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ExcludedPath.java:28]: Now we have this constructor , what is the benefit of having default constructor and setPath() method explicitly .

File: sdk/cosmos/azure-cosmos-benchmark/src/test/java/com/azure/cosmos/benchmark/ReadMyWritesConsistencyTest.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-benchmark/src/test/java/com/azure/cosmos/benchmark/ReadMyWritesConsistencyTest.java]: Sure can, has changed across all the places.

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/models/IncludedPathTest.java (hunk 2 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/models/IncludedPathTest.java:16]: added

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ExcludedPath.java (hunk 3 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ExcludedPath.java:28]: Good question, I am not so sure.

I think if path is required for includedPath, and we do not have any default path value, then should remove the default constructor.  (I guess same question for excludedPath).

@moderakh @kushagraThapar Any thoughts for this one?

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ExcludedPath.java:28]: Good point, I say we remove default constructor, and just keep the constructor which takes `String path`

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/models/IncludedPathTest.java [hunk 4]
```diff
@@ -8,33 +8,25 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import static org.assertj.core.api.Assertions.assertThat;
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/models/IncludedPathTest.java:11]: please add back static import so we don't repeat Assertions.

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/models/IncludedPathTest.java [hunk 5]
```diff
@@ -8,33 +8,25 @@
...
+        String path = "/*";
+        IncludedPath includedPath = new IncludedPath(path);
+
+        List<Index> indexes = new ArrayList<>();
+        indexes.add(Index.range(DataType.STRING, -1));
+        indexes.add(Index.range(DataType.NUMBER, -1));
+        includedPath.setIndexes(indexes);
+
+        List<Index> includedPathIndexes = new ArrayList<>(includedPath.getIndexes());
+        Assertions.assertThat(includedPathIndexes).hasSize(2);
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/models/IncludedPathTest.java:24]: please have static import so we don't repeat Assertions.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ExcludedPath.java (hunk 3 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ExcludedPath.java:28]: what about setPath()? probably that shouldn't be public anymore now as path is set in the constructor.

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ExcludedPath.java:28]: We can leave the `setPath()` API as it is, doesn't harm to have the API - and will be in sync with `setIndexes()` API.

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ExcludedPath.java:21]: Removed default constructor from includedPath and excludedPath

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/models/IncludedPathTest.java (hunk 5 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/models/IncludedPathTest.java]: added static import

**Pull request merged**